tasks

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+dev_image=docker-cli-dev
+linter_image=docker-cli-lint
+cross_image=docker-cli-cross
+
+
+function run_task {
+    local task=$1
+
+    if [[ "$task" = *shell ]]; then
+        local cmd=sh
+    else
+        local cmd="make $task"
+    fi
+    
+    case $task in
+        lint|lint-shell)
+            docker_build_and_run "$linter_image" ./dockerfiles/Dockerfile.lint "$cmd"
+            ;;
+        cross|dynbinary|cross-shell)
+            docker_build_and_run "$cross_image" ./dockerfiles/Dockerfile.cross "$cmd"
+            ;;
+        *)
+            docker_build_and_run "$dev_image" ./dockerfiles/Dockerfile.dev "$cmd"
+            ;;
+    esac
+}
+
+function docker_build_and_run {
+    local image=$1
+    local dockerfile=$2
+    local cmd=$3
+    local dockerfile_source=
+    local cidfile=
+    local remove="--rm"
+    local mounts="-v $PWD:/go/src/github.com/docker/cli"
+    if [ -t 1 ] ; then local use_tty="-ti"; else local use_tty=""; fi
+
+    if [ -n "${DOCKER_HOST:-}" ] || [ -n "${NO_BINDMOUNT:-}" ]; then
+        dockerfile_source="$(cat $dockerfile <(echo COPY . .))"
+        mounts=""
+        dockerfile="-"
+    fi
+
+    if [ -n "${CID_FILENAME:-}" ]; then
+        cidfile="--cidfile $CID_FILENAME"
+        remove=
+    fi
+
+    echo "$dockerfile_source" | docker build -t "$image" -f "$dockerfile" .
+    docker run $remove $cidfile $use_tty $mounts "$image" $cmd
```

Comment:
while you're at it, this need to export `VERSION`, `GITCOMMIT`. `BUILDTIME`  and `LDFLAGS` from the env (to match what's in `scripts/build/.variables`)